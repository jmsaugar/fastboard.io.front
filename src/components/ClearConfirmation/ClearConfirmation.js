import React from 'react';
import PropTypes from 'prop-types';
import { useTranslation } from 'react-i18next';

import Button from '../Button';
import {
  SCard, SHeader, SErrorIcon, SContent, SFooter,
} from './styled';

const ClearConfirmation = ({ onConfirm, onCancel }) => {
  const { t } = useTranslation('board');

  return (
    <SCard>
      <SHeader>
        <SErrorIcon />
        {t('clearConfirmation.title')}
      </SHeader>
      <SContent>
        {t('clearConfirmation.about')}
      </SContent>
      <SFooter>
        <Button type="secondary" onClick={onCancel}>
          {t('clearConfirmation.cancel')}
        </Button>
        <Button type="primary" onClick={onConfirm}>
          {t('clearConfirmation.accept')}
        </Button>
      </SFooter>
    </SCard>
  );
};

ClearConfirmation.propTypes = {
  onConfirm : PropTypes.func.isRequired,
  onCancel  : PropTypes.func.isRequired,
};

export default ClearConfirmation;
