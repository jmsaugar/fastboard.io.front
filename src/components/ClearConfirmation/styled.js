import styled, { css, down } from '@xstyled/styled-components';
import { ExclamationCircle as ErrorIcon } from '@styled-icons/fa-solid/ExclamationCircle';

import Card from '../Card';

export const SCard = styled(Card)`
  display: flex;
  flex-direction: column;
  justify-content: center;
  ${down('md', css`text-align: center;`)}
`;

export const SHeader = styled.h2`
  margin: 0;
  color: warning;
  font-size: xlg;
`;

export const SErrorIcon = styled(ErrorIcon)`
  width: 34px;
  height: 34px;
  margin-top: -4px;
  margin-right: sm;
`;

export const SContent = styled.div`
  margin: default 0;
`;

export const SFooter = styled.div`
  display: flex;
  flex-direction: row;
  justify-content: flex-end;

  button + button {
    margin-left: sm;
  }

  ${down('md', css`button { width: 100%; }`)}
`;
