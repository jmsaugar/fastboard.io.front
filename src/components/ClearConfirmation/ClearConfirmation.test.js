import React from 'react';
import { fireEvent } from '@testing-library/react';

import { render } from '#devTools/tests/utils';

import ClearConfirmation from './ClearConfirmation';

describe('Component : ClearConfirmation', () => {
  test('Is visible and reacts to clear action', () => {
    const onConfirmFn = jest.fn();

    const { getByText } = render(
      <ClearConfirmation
        onConfirm={onConfirmFn}
        onCancel={jest.fn()}
      />,
    );

    expect(getByText('clearConfirmation.about')).toBeInTheDocument();
    fireEvent.click(getByText('clearConfirmation.accept'));
    expect(onConfirmFn).toHaveBeenCalledTimes(1);
  });

  test('Is visible and reacts to cancel action', () => {
    const onCancelFn = jest.fn();

    const { getByText } = render(
      <ClearConfirmation
        onConfirm={jest.fn()}
        onCancel={onCancelFn}
      />,
    );

    expect(getByText('clearConfirmation.about')).toBeInTheDocument();
    fireEvent.click(getByText('clearConfirmation.cancel'));
    expect(onCancelFn).toHaveBeenCalledTimes(1);
  });

  test('Snapshot', () => {
    const { asFragment } = render(
      <ClearConfirmation
        onConfirm={jest.fn()}
        onCancel={jest.fn()}
      />,
    );

    const componentRender = asFragment();
    expect(componentRender).toMatchSnapshot();
  });
});
