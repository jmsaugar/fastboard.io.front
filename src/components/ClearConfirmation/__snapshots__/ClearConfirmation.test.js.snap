// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component : ClearConfirmation Snapshot 1`] = `
<DocumentFragment>
  <div
    class="sc-qZtCU XqBnS sc-pzMyG gEsWbO"
  >
    <h2
      class="sc-pHIBf bQYmRH"
    >
      <svg
        aria-hidden="true"
        class="StyledIconBase-ea9ulj-0 jVnGjM sc-pQEbo eheZPo"
        fill="currentColor"
        focusable="false"
        viewBox="0 0 512 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M504 256c0 136.997-111.043 248-248 248S8 392.997 8 256C8 119.083 119.043 8 256 8s248 111.083 248 248zm-248 50c-25.405 0-46 20.595-46 46s20.595 46 46 46 46-20.595 46-46-20.595-46-46-46zm-43.673-165.346l7.418 136c.347 6.364 5.609 11.346 11.982 11.346h48.546c6.373 0 11.635-4.982 11.982-11.346l7.418-136c.375-6.874-5.098-12.654-11.982-12.654h-63.383c-6.884 0-12.356 5.78-11.981 12.654z"
          fill="currentColor"
        />
      </svg>
      clearConfirmation.title
    </h2>
    <div
      class="sc-pYA-dN jOHHWI"
    >
      clearConfirmation.about
    </div>
    <div
      class="sc-oUcyK gRJMGp"
    >
      <button
        class="sc-qQYBZ kBBHFN"
        type="secondary"
      >
        clearConfirmation.cancel
      </button>
      <button
        class="sc-qQYBZ bNCOou"
        type="primary"
      >
        clearConfirmation.accept
      </button>
    </div>
  </div>
</DocumentFragment>
`;
